### PR TITLE
Add preamble section

### LINKED ISSUE #44: Add section after imports and before dependencies
## Describe

When generating classes with a custom template, I need to add some extra initialisation matter after the imports and before the list of classes, e.g.

```py
from pydantic import Field
import pydantic
from typing import Any, Dict, List, Literal, Optional, Union
import dataclasses
[ ... ]

# some additional initialisations here
dataclasses._init_fn = dataclasses_init_fn_with_kwargs

class WeakRefBaseModel(pydantic.BaseModel):
    __slots__ = '__weakref__'

# class list starts here
```

Do you have any objections to the module being extended in this manner?

### PR DESCRIPTION
**Pull request by ialarmedalien**
This PR enables extra matter to be added between the imports and the generated class list using a `--preamble` command line parameter.

I also fixed a few minor grammar bits and my IDE automatically removes trailing whitespace.

Closes #44

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -405,9 +405,45 @@ class Run(BaseModel):
...
+
+<details><summary>----- Show -----</summary>
+<p>
+A simple example to demonstrate adding extra code before the class list.
+
+```sh
+json2models -f pydantic --preamble "# set up defaults
+USERNAME = 'user'
+SERVER_IP = '127.0.0.1'
+" -m Swagger testing_tools/swagger.json
```
Comment (ialarmedalien) [inline README.md:418]: not very easy to show multi-line arguments with the current readme structure so I added this extra example

File: README.md [hunk 2]
```diff
@@ -464,61 +500,71 @@ Arguments:
...
+* `--preamble` - Additional material to be
+    * **Format**: `--preamble "<formatted python code string to be added after module imports>"`
+    * **Example**:
+  ```sh
+  --preamble "# set up defaults
+  USERNAME = 'user'
+  SERVER = '127.0.0.1'"
+  ```
+    * **Optional**
+
```
Comment (ialarmedalien) [inline README.md:518]: all the rest of the diffs below are whitespace removal

File: json_to_models/cli.py [hunk 3]
```diff
@@ -102,11 +102,12 @@ def parse_args(self, args: List[str] = None):
         code_generator_kwargs_raw: List[str] = namespace.code_generator_kwargs
         dict_keys_regex: List[str] = namespace.dict_keys_regex
         dict_keys_fields: List[str] = namespace.dict_keys_fields
+        preamble: str = namespace.preamble
 
-        self.validate(models, models_lists, merge_policy, framework, code_generator)
```
Comment (ialarmedalien) [inline json_to_models/cli.py:106]: `models` is not used

File: json_to_models/cli.py [hunk 4]
```diff
@@ -140,11 +144,10 @@ def version_string(self):
             '"""\n'
         )
 
-    def validate(self, models, models_list, merge_policy, framework, code_generator):
+    def validate(self, models_list, merge_policy, framework, code_generator):
         """
         Validate parsed args
 
-        :param models:  List of pairs (model name, list of filesystem path)
```
Comment (ialarmedalien) [inline json_to_models/cli.py:147]: unused => removed it

File: test/test_cli/test_script.py [hunk 5]
```diff
@@ -90,7 +90,8 @@ def test_help():
 ]
 
 
-def _validate_result(proc: subprocess.Popen, output=None, output_file: Path = None) -> Tuple[str, str]:
+def execute_test(command, output_file: Path = None, output=None) -> Tuple[str, str]:
```
Comment (ialarmedalien) [inline test/test_cli/test_script.py:93]: Small refactor here to reduce repeated code in the tests in this file

File: test/test_cli/test_script.py [hunk 6]
```diff
@@ -108,73 +109,76 @@ def _validate_result(proc: subprocess.Popen, output=None, output_file: Path = No
...
+
+    print(stdout)
+    return stdout
 
 
 @pytest.mark.parametrize("command", test_commands)
 def test_script(command):
-    proc = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-    stdout, stderr = _validate_result(proc)
-    print(stdout)
```
Comment (ialarmedalien) [inline test/test_cli/test_script.py:118]: these lines are combined in `execute_test()`

**Comment by bogdandm** [comment]
Thank you for this pr, seems good except for this small formatting bug.

**Comment by ialarmedalien** [comment]
Do you want me to add a note to the changelog and bump the version too?

**Comment by bogdandm** [comment]
No, the changelog is generated automatically and the version is updated with a single command (and the release is done by the github action). So I'll do it myself.

**Comment by ialarmedalien** [comment]
@bogdandm change pushed

**Review by bogdandm — APPROVED** [review]

**Comment by bogdandm** [comment]
Test fails now :( AttributeError: 'NoneType' object has no attribute 'strip'

**Comment by ialarmedalien** [comment]
Should be fixed now, with another test added to ensure that the `strip()` operation works.

**Comment by bogdandm** [comment]
0.2.6 released https://github.com/bogdandm/json2python-models/releases/tag/v0.2.6

**Pull request merged**